### ISSUE TITLE
500 HTTP Error on OpenAPI page

### ISSUE THREAD

**Issue by drnextgis** [stac-utils/stac-fastapi-pgstac#173]
I recently pulled the latest version of stac-fastapi-pgstac and started it using the following command:
```
$ docker compose up app
```
However, when I navigate to http://0.0.0.0:8082/api.html, I encounter a 500 HTTP error.

Here's the stack trace:
```
...
stac-fastapi-pgstac  |   File "/usr/local/lib/python3.11/site-packages/pydantic/json_schema.py", line 1569, in resolve_ref_schema
stac-fastapi-pgstac  |     raise RuntimeError(f'Cannot update undefined schema for $ref={ref}')
stac-fastapi-pgstac  | RuntimeError: Cannot update undefined schema for $ref=#/components/schemas/stac_fastapi__pgstac__extensions__query__Operator-Input__1
```

**Comment by drnextgis** [comment]
The issue is linked to Pydantic version 2.10.x, as discussed in this thread: https://github.com/pydantic/pydantic/issues/10960

**Comment by vincentsarago** [comment]
@drnextgis was this fixed upstream ?

**Comment by drnextgis** [comment]
Yes, it was fixed in https://github.com/pydantic/pydantic/releases/tag/v2.10.3. Should we update the [dependencies](https://github.com/stac-utils/stac-fastapi-pgstac/blob/713505947bbea6b25f14a182d6d28000875b05ab/setup.py#L11) to reflect this?

**Comment by vincentsarago** [comment]
I don't think we need to do anything, or we could exclude `2.10.0|2.10.1|2.10.2` but I think there are low chances that users have pydantic pinned to those specific version
